spock-core/src/main/java/org/spockframework/runtime/SpockExecutionContext.java

```diff
@@ -48,6 +60,11 @@ private SpockExecutionContext setSpec(SpecInfo spec) {
   }
 
   private SpockExecutionContext setCurrentFeature(FeatureInfo currentFeature) {
+    if (this.currentFeature != currentFeature) {
+      hadSuccess.set(false);
+      abortions.clear();
```

Comment:
The context is designed to be cloned, not modified without being copied.

Reply:
Well, with the current tactic it wouldn't work otherwise, as I needed the results of all iterations in the feature after the iterations were run.
But if we store it in the iteration node as suggested above, this would also be resolved immediately.